lib/sentry/context.ex

```diff
@@ -57,8 +57,9 @@ defmodule Sentry.Context do
     Process.get(@process_dictionary_key) || %{}
   end
 
-  def add_breadcrumb(map) when is_map(map) do
-    map = Map.put_new(map, "timestamp", Sentry.Util.unix_timestamp())
+  @spec add_breadcrumb(Sentry.Context.BreadCrumb.t) :: nil
+  def add_breadcrumb(%Sentry.Context.BreadCrumb{} = map) do
```

Comment:
After some thinking, I'm not sure if we want to pattern match on the struct, as it will break backwards compatibility since people are using maps currently.  What do you think would be the best way to work around that?